activesupport/lib/active_support/encryptor/jwe.rb

```diff
@@ -0,0 +1,157 @@
+require 'base64'
+require 'openssl'
+require 'active_support/core_ext/hash/keys'
+
+module ActiveSupport
+  class MessageEncryptor
+    class JWE # :nodoc:
+      attr_reader :cipher
+
+      def initialize(**options)
+        @options = options
+        @secret = @options[:secret]
+        @alg = key_encryption_algorithm
+        @cipher = encryption_algorithm
+        @enc = CIPHER_ENC_PAIRS[@cipher]
+        @serializer = @options[:serializer] || Marshal
+      end
+
+      def encrypt(value)
```

Comment:
Both encrypt and decrypt are still extremely tough to understand. We need to find a way to make them simpler.